### PR TITLE
Rasterlite2 styles

### PR DESCRIPTION
**Pull request by mj10777**
Reverted rasterlite2_styles back to original state. 

New is:
- GeopaparazziOverlay.drawOverlayBitmap 
  - splitted into functions for each type being drawn(Geometries,Ways,Items,GpsLog,GpsMarker)

### CODE REVIEW

File: geopaparazzi.app/src/eu/hydrologis/geopaparazzi/maps/overlays/GeopaparazziOverlay.java [hunk 1]
```diff
@@ -506,11 +514,62 @@ protected void drawOverlayBitmap(Canvas canvas, Point drawPosition, Projection p
...
+     * @param canvas
+     * @param drawPosition
+     * @param projection
+     * @param drawZoomLevel
+     * @return nothing
+     */
+    private void drawGpsLog(Canvas canvas, Point drawPosition, Projection projection, byte drawZoomLevel) {
+     if (stopDrawing()) {
+        // stop working
+        return;
```
Comment (moovida) [inline geopaparazzi.app/src/eu/hydrologis/geopaparazzi/maps/overlays/GeopaparazziOverlay.java]: This does no longer work as it should. This return was meant for the method it was in before.

File: geopaparazzilibrary/src/eu/geopaparazzi/library/util/LibraryConstants.java [hunk 2]
```diff
@@ -40,6 +40,17 @@
...
+     * - Images (Photos) imported to RasterLite2 will recieve the srid=-1
+     * - since this value cannot be transformed to Wsg84
+     * Since 2012, Spatialite contains historical 'DHDN / Soldner *' values that have been in use since 1879
+     * --> SELECT * FROM spatial_ref_sys WHERE auth_name = 'mj10777.de';
+     * I also 'sneaked' in an  imaginary Projection based on the Brandenburg Gate, Berlin, Germany at point 0,0
+     * - this (otherwise unused) Projection will be reserved to render Photo-Images
+     * -- since every Spatialite-Database with a full 'spatial_ref_sys',since 2012, contains this Projection.
+     */
+    public final String SRID_BRANDENBURGER_TOR_187999 = "187999"; //$NON-NLS-1$
+    /**
```
Comment (moovida) [inline geopaparazzilibrary/src/eu/geopaparazzi/library/util/LibraryConstants.java]: I do not want to have this here. It is too specific. 
Why would this be necessary? The Mercator 3957 already is a metric one.

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/DaoSpatialite.java [hunk 3]
```diff
@@ -436,9 +439,51 @@ public static void updateFeatureGeometry(String id, Geometry geometry, String ge
         sbIn.append(SpatialiteUtilities.SPATIALTABLE_ID_FIELD).append("=");
         sbIn.append(id);
         String insertQuery = sbIn.toString();
-        database.exec(insertQuery, null);
+        dbSpatialite.exec(insertQuery, null);
+    }
+    public static double[] longLat2Srid(Database dbSpatialite, double lon, double lat, String s_srid)  throws Exception {
```
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/DaoSpatialite.java]: Why is this method necessary? A javadoc would help

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java [hunk 4]
```diff
@@ -187,19 +190,289 @@ public static boolean hasRasterLiteSupport(){
...
+        // SELECT
+        // RL2_GetMapImageFromVector('1890.berlin_postgrenzen',BuildMBR(20800.0,22000.0,24000.0,19600.0),1200,1920,'default','image/png','#ffffff',0,0,1);
+        String mbr = mbrSb.toString();
+        StringBuilder qSb = new StringBuilder();
+        qSb.append("SELECT RL2_GetMapImageFromVector('");
+        qSb.append(coverageName);
+        qSb.append("',");
+        qSb.append(mbr);
+        qSb.append(",");
+        qSb.append(Integer.toString(width));
```
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java]: on the stringbuilder you do not need to make Integer.toString. You can simply pass the integer.

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java [hunk 5]
```diff
@@ -187,19 +190,289 @@ public static boolean hasRasterLiteSupport(){
...
+     * @return zoom_levels
+     */     
+    public static int[] rl2_calculate_zoom_levels(Database dbSpatialite, String coverageName, String s_tile_size,int i_srid) {
+        HashMap<Integer, Double> zoom_levels = new HashMap<Integer, Double>();  
+        // TODO: remove to Mercator-specific definitions
+        // 'WGS 84 / World Mercator' (3395) - UNIT=Meters            
+        // Zoom-Level, Width in Meters of a 256x256 Tile
+        // Value returned by gdalinfo for a geo-referenced Tile
+        // values based on the position: 13.37771496361961 52.51628011262304 [Brandenburg Gate, Berlin, Germany]
+        zoom_levels.put( 0,40032406.294);
```
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java]: Don't we have the list of zoomlevels already somewhere else? I think this here might be a duplication.

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/databasehandlers/MbtilesDatabaseHandler.java [hunk 6]
```diff
@@ -152,18 +152,61 @@ public boolean isValid() {
...
+     * <li>vector_key[mbtiles]='databaseFileNameNoExtension;0;MBTiles;title;abstract' = length=5[0-4]</li>
+     * <li>vector_data[mbtiles]='minZoom;maxZoom;srid;0' = length=4[0-3]</li>
+     * <li>vector_extent[mbtiles]='0;extent_0-6;?,?,?;getDatabasePath()' = length=4[0-3]</li>
+     * </ol>
+     * Note: title and abstract (short/long descriptions) replace ';' with '-'
+     * <p> vector_key/data documetation : 20150718
+     * 
+     * @return list of RasterTabes.
+     */     
     public List<SpatialRasterTable> getSpatialRasterTables( boolean forceRead ) throws Exception {
```
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/databasehandlers/MbtilesDatabaseHandler.java]: From here I can't see it. Is this method available from an interface or abstract class? If yes, the javadoc should be placed only there. I have the feeling I have seen that javadoc a few times already. Is it the same and can be centralized for better maintainance?

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/util/SpatialiteUtilities.java [hunk 7]
```diff
@@ -43,12 +43,74 @@
...
      * Name/path separator for spatialite table names.
      */
     public static final String UNIQUENAME_SEPARATOR = "#"; //$NON-NLS-1$
 
+    /**
+     * Active Connection to a  Spatialite Database
+    * <p/>
+     * <p>Indented for general use for Transformation etc.
+     */
+    private static Database dbSpatialite=null;
```
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/util/SpatialiteUtilities.java]: I am quite sure I don't want these to be STATIC. Using complex objects as static in Android is way to risky for memory leaks. This NEEDS to be changed most probably. Same below for memory db.

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/util/SpatialiteUtilities.java [hunk 8]
```diff
@@ -43,12 +43,74 @@
...
+    /**
+     * Active Connection to a  Spatialite Database
+    * <p/>
+     * <p>Indented for general use for Transformation etc.
+     */
+    private static Database dbMemSpatialite=null;
```
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/util/SpatialiteUtilities.java]: Do we really want to support memory databases? I can't find a good use for that on android. Also because the OS might kill your app while you are logging due to memory issues. 
Should be better removed?

Comment (mj10777) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/util/SpatialiteUtilities.java]: 2015-09-19 14:30 GMT+02:00 Andrea Antonello <notifications@github.com>:

> In
> geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/util/SpatialiteUtilities.java
> <https://github.com/geopaparazzi/geopaparazzi/pull/287#discussion_r39919170>
> :
>
> >      /**
> >       * Name/path separator for spatialite table names.
> >       */
> >      public static final String UNIQUENAME_SEPARATOR = "#"; //$NON-NLS-1$
> >
> > +    /**
> > +     * Active Connection to a  Spatialite Database
> > +    * <p/>
> > +     * <p>Indented for general use for Transformation etc.
> > +     */
> > +    private static Database dbSpatialite=null;
> > +    /**
> > +     * Active Connection to a  Spatialite Database
> > +    * <p/>
> > +     * <p>Indented for general use for Transformation etc.
> > +     */
> > +    private static Database dbMemSpatialite=null;
>
> Do we really want to support memory databases? I can't find a good use for
> that on android. Also because the OS might kill your app while you are
> logging due to memory issues.
> Should be better removed?
>
This was intended for the case where no active spatialite connection exists
* transforming the position

The static one would be set by the first read spatialite connection
* a memory only if none exist

was the idea.

I also don't like the idea.

If the Database used in 'GPApplication'
* used a spatialite connection (jsqlite.Database)
   * then that could be used

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/DaoSpatialite.java (hunk 3 above)
Comment (mj10777) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/DaoSpatialite.java]: 2015-09-19 14:09 GMT+02:00 Andrea Antonello <notifications@github.com>:

> In
> geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/DaoSpatialite.java
> <https://github.com/geopaparazzi/geopaparazzi/pull/287#discussion_r39919038>
> :
>
> > @@ -436,9 +439,51 @@ public static void updateFeatureGeometry(String id, Geometry geometry, String ge
> >          sbIn.append(SpatialiteUtilities.SPATIALTABLE_ID_FIELD).append("=");
> >          sbIn.append(id);
> >          String insertQuery = sbIn.toString();
> > -        database.exec(insertQuery, null);
> > +        dbSpatialite.exec(insertQuery, null);
> > +    }
> > +    public static double[] longLat2Srid(Database dbSpatialite, double lon, double lat, String s_srid)  throws Exception {
>
> Why is this method necessary? A javadoc would help
>
Yes, is missing,
* is to transform the screen position to the srid of the map

for AbstractSpatialTable:

    /*
     * Convert a longitude and latitude to the table's
     * original SRID.
     */
    public abstract double[] longLat2Srid(double lon, double lat);

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/databasehandlers/MbtilesDatabaseHandler.java (hunk 6 above)
Comment (mj10777) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/databasehandlers/MbtilesDatabaseHandler.java]: 2015-09-19 14:23 GMT+02:00 Andrea Antonello <notifications@github.com>:

> In
> geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/databasehandlers/MbtilesDatabaseHandler.java
> <https://github.com/geopaparazzi/geopaparazzi/pull/287#discussion_r39919113>
> :
>
> > +     * <li>5.7: defaultZoom - can be user defined</li>
> > +     * <li>6:last_verified - tileQuery as '?,?,?'</li>
> > +     * <li>7:getDatabasePath()</li>
> > +     * <li>8-?:not used</li>
> > +     * </ol>
> > +     * <p>vector_key and vector_value
> > +     * <ol>
> > +     * <li>vector_key[mbtiles]='databaseFileNameNoExtension;0;MBTiles;title;abstract' = length=5[0-4]</li>
> > +     * <li>vector_data[mbtiles]='minZoom;maxZoom;srid;0' = length=4[0-3]</li>
> > +     * <li>vector_extent[mbtiles]='0;extent_0-6;?,?,?;getDatabasePath()' = length=4[0-3]</li>
> > +     * </ol>
> > +     * Note: title and abstract (short/long descriptions) replace ';' with '-'
> > +     * <p> vector_key/data documetation : 20150718
> > +     *
> > +     * @return list of RasterTabes.
> > +     */
> >      public List<SpatialRasterTable> getSpatialRasterTables( boolean forceRead ) throws Exception {
>
> From here I can't see it. Is this method available from an interface or
> abstract class? If yes, the javadoc should be placed only there. I have the
> feeling I have seen that javadoc a few times already. Is it the same and
> can be centralized for better maintainance?
>

The main documentation is in
* AbstractSpatialTable.parse_vector_key_value
   * the only place they are now being **read** / parsed

They are **created** in different places (above mbtiles)
* there I added it also to show how they were used

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java (hunk 5 above)
Comment (mj10777) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java]: 2015-09-19 14:17 GMT+02:00 Andrea Antonello <notifications@github.com>:

> In
> geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java
> <https://github.com/geopaparazzi/geopaparazzi/pull/287#discussion_r39919086>
> :
>
> >                      }
> >              }
> >          }
> >          return ba_image;
> >      }
> > +    /**
> > +     *  HashMap<Integer, Double> of Zoom-Levels 0-30
> > +     *  Zoom-Level, Width in World-Mercator-Meters of a 256x256 Tile
> > +     *  - 'WGS 84 / World Mercator' (3395) - UNIT=Meters
> > +     * @return zoom_levels
> > +     */
> > +    public static int[] rl2_calculate_zoom_levels(Database dbSpatialite, String coverageName, String s_tile_size,int i_srid) {
> > +        HashMap<Integer, Double> zoom_levels = new HashMap<Integer, Double>();
> > +        // TODO: remove to Mercator-specific definitions
> > +        // 'WGS 84 / World Mercator' (3395) - UNIT=Meters
> > +        // Zoom-Level, Width in Meters of a 256x256 Tile
> > +        // Value returned by gdalinfo for a geo-referenced Tile
> > +        // values based on the position: 13.37771496361961 52.51628011262304 [Brandenburg Gate, Berlin, Germany]
> > +        zoom_levels.put( 0,40032406.294);
>
> Don't we have the list of zoomlevels already somewhere else? I think this
> here might be a duplication.
>
I did look for that, but found none
* these I worked out a few years ago and they work correctly

I have asked Sandro to build something like this into rasterlite2, so that
it can be quried
* but until November there will be no reply

File: geopaparazzi.app/src/eu/hydrologis/geopaparazzi/maps/overlays/GeopaparazziOverlay.java (hunk 1 above)
Comment (moovida) [inline geopaparazzi.app/src/eu/hydrologis/geopaparazzi/maps/overlays/GeopaparazziOverlay.java]: @mj10777 any comment here?

File: geopaparazzilibrary/src/eu/geopaparazzi/library/util/LibraryConstants.java (hunk 2 above)
Comment (moovida) [inline geopaparazzilibrary/src/eu/geopaparazzi/library/util/LibraryConstants.java]: Same here, please comment or change.

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/DaoSpatialite.java (hunk 3 above)
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/DaoSpatialite.java]: But the srid of the map can only be 3857, right? Why is this necessary?
I would love to see generic srids involved, but libraries that start from OSM tiles only support 3857

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java (hunk 4 above)
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java]: Would you mind to apply those changes or comment?

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java (hunk 5 above)
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java]: Ok, it might be a good idea to put the zoomlevels in a central place. It seems like a reusable information to me. I will do that after the merge of the pull request.

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/databasehandlers/MbtilesDatabaseHandler.java (hunk 6 above)
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/databasehandlers/MbtilesDatabaseHandler.java]: What I mean is that I see a lot of what seems to be duplicated javadoc, which is complex and easily outdated (it has been often in the past). So I wonder where that should go.

For sure a method that implements an interface method should have **NO** own javadoc, because the need for it would break the reason to have an interface. The javadoc of the interface should be enough.

So please move that javadoc to a more appropriate place (as normal doc inside the method?). 
I really want to try to fix the use of documentation and do it properly the java way.

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/util/SpatialiteUtilities.java (hunk 7 above)
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/util/SpatialiteUtilities.java]: Have you checked this? This is an important one.

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/util/SpatialiteUtilities.java (hunk 8 above)
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/util/SpatialiteUtilities.java]: Hmm, that is a quite interesting usecase, ok, let's keep it in.

**Comment by moovida** [comment]
@mj10777 just a note that for this to be merged in we will have to agree on all the above comments. Thank you.

File: geopaparazzilibrary/src/eu/geopaparazzi/library/util/LibraryConstants.java (hunk 2 above)
Comment (mj10777) [inline geopaparazzilibrary/src/eu/geopaparazzi/library/util/LibraryConstants.java]: Sorry, I answered these as a reply on the first day.
* but it seems it has to be answered here

The worked out values are in 3395 (WGS 84 / World Mercator)
* there was a reason for not using 3857 (it was worked out many years ago), but I don't remember the reason now
* 0,0 is the equator, where the width is the smallest

For the resolution if non-georeferenced Images (srid=-1)
* which always has a 0,0
   * it brings better results

Mapsforge does not support srid: -1 images, which can be stored in a RasterLite2 Database
* so the received wsg84 values must be transformed into something valid

The reason for its use as a const value, it because it is used in two functions

File: geopaparazzi.app/src/eu/hydrologis/geopaparazzi/maps/overlays/GeopaparazziOverlay.java (hunk 1 above)
Comment (mj10777) [inline geopaparazzi.app/src/eu/hydrologis/geopaparazzi/maps/overlays/GeopaparazziOverlay.java]: It was placed before the gps-code in the original function
* I assumed it should break at this point if it should stop drawing

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/DaoSpatialite.java (hunk 3 above)
Comment (mj10777) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/DaoSpatialite.java]: this is for the implementation of:

AbstractSpatialTable:
    /*
     * Convert a longitude and latitude to the table's
     * original SRID.
     */
    public abstract double[] longLat2Srid(double lon, double lat);

to show the present position on the screen
* which at the moment is only wsg84

The srid of the table being used would be sent to this function
* with the position returned in that srid
   * which could be placed then on the screen

It can also be used for any other position transformation.

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java (hunk 4 above)
Comment (mj10777) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java]: You mean just removing `Integer.toString(` ?
* if yes, I will

The GetMapImageFromVector logic is still in work
* so more work is needed

But if that is a yes, I will remove those tomorrow.

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java (hunk 5 above)
Comment (mj10777) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java]: I was looking for a place
* but was not sure where

I am presently working on a cutline and crop to cutline for RasterLite2
* that I will be submitting to Sandro in the next week or so

I have also added the functionality to retrieve suggested Zoom-Levels
* if Sandro accepts this
   * then `rl2_calculate_zoom_levels` will no longer be needed
      * and can then be removed

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/DaoSpatialite.java (hunk 3 above)
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/DaoSpatialite.java]: Ah, ok, now I get it. Lat long can be transformed to that layer's srid. Ok, nice.

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java (hunk 4 above)
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java]: Ok, no, if it is still in work, you can do it when doing the work you are planning to. I just don't wanted to forget that myself. This is ok for me then.

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java (hunk 5 above)
Comment (moovida) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java]: Ok, then for now let's leave it as you have it here. When your changes get into rasterlite, we can "cleanup".

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/util/SpatialiteUtilities.java (hunk 8 above)
Comment (mj10777) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/util/SpatialiteUtilities.java]: If the 'GPApplication' has an active spatialite connection
* then these would **not** be needed

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java (hunk 4 above)
Comment (mj10777) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java]: I will do it tomorrow, so that it is done

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/databasehandlers/MbtilesDatabaseHandler.java (hunk 6 above)
Comment (mj10777) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/databasehandlers/MbtilesDatabaseHandler.java]: It **needs** to be in
* `parse_vector_key_value`
   * because there it is parsed for **all** formats

The others were only for the specific format
* being created in that function as a reference to the exact syntax
   * if something must be changed
      * so it is not strictly needed

When I make a change (as in RasterLite2 yesterday)
* I changed the doc there
   * copied it over to `parse_vector_key_value`
      * and adaped the parsing code

File: geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/util/SpatialiteUtilities.java (hunk 7 above)
Comment (mj10777) [inline geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/util/SpatialiteUtilities.java]: See below at the comments with  'GPApplication'

**Comment by mj10777** [comment]
I have answered a lot of these
* could you make a new list of what point are not agreed to please
   * getting slightly lost ..

**Comment by moovida** [comment]
I agree with you, it is a pain read those here :-) 
I have no idea about how to create a new list. But we are almost there. My last doubt is the static reference to the database. I really don't like that, I know that it will give trouble. I will think about it.

**Comment by mj10777** [comment]
if I have understood 'GPApplication' correctly
* the GetDatabase()
   * is the connection that exists for the gps and log entries of the Application

If this was a spatialite connection
* that can be used for quick general non-database spatial queries
   * the static databases can be removed

the static database
* would have received the first active spatialite connection

It was only because there was no garantie that one existed
* the user uses on .map files
   * that the memory was planed as a fallback

With a GPApplication.GetDatabase() with a spatialite connection
* all of that can be removed, since no longer needed for anything

**Comment by mj10777** [comment]
Add it to the bottom of this one and I will see it

**Comment by moovida** [comment]
> if I have understood 'GPApplication' correctly
> 
> the GetDatabase()
> is the connection that exists for the gps and log entries of the Application
> If this was a spatialite connection
> 
> that can be used for quick general non-database spatial queries
> the static databases can be removed
> the static database
> 
> would have received the first active spatialite connection
> It was only because there was no garantie that one existed
> 
> the user uses on .map files
> that the memory was planed as a fallback
> With a GPApplication.GetDatabase() with a spatialite connection
> 
> all of that can be removed, since no longer needed for anything

I would be even keen to create a singleton to keep a default db connection, if that is necessary. 
The problem here is that it can't be in GPApplication, because the library project doesn't know anything about spatialite... and it can't be in Geopaparazzi, since then the spatialite project could not make use of it (I assume ti would need to). 
So it would need to be in the spatialite project. Maybe a SpatialiteApplication class? Not sure though...

**Comment by mj10777** [comment]
I have completed changes to RasterLite2 and submitted a patch to Sandro
* when he will deal with this in another question 
   * until Spatialite-Topology has been completed unlikly

This patch will include a new command called
* RL2_GetRasterCoverageDefaults

This function will return the information presently calculated
* rl2_calculate_zoom_levels

I Sandro accepts this, then
* rl2_calculate_zoom_levels and
* LibraryConstants.SRID_BRANDENBURGER_TOR_187999
could then be removed.

I would suggest, at least temporary, accepting these two portions
* so this branch can be merged

My suggestion to Sandro is that he deal with this before continuing the RL2 development
* if this is done, the I would create a Library based on these changes
   * add RL2_GetRasterCoverageDefaults (which I have tested in my version)
   * remove the two above meantime

The final result would then be a cleaned up version
* ready to be adapted to a final Rasterlite2 version

**Comment by moovida** [comment]
@mj10777 , I will now start to merge in this big chunk of code in and then I will have to fix a whole pile of stuff, which I am not happy about. You didn't fix some of the things I asked you to, and some of them might cause memory leaks. Honestly it is not the first time, but it needs to be the last. 
From now on please use java coding conventions else it gets too hard for me to read through the code. Honestly, it is for example a while that I want to add the possibility to add tiles/database source dynamically, but I never figured out how those parts work. This is not good.

**Comment by mj10777** [comment]
At some point, some portions of this merge that were not taken over
* should be reconsidered

Mainly the area where the simplification of the parameters was changed.

The goal was for each source to collect the information needed
* sending it off in an Array
   * which would then be interpreted in one place
      * AbstractSpatialTable.parse_vector_key_value

---
This was done mainly to support
* Spatial-Topopoly
* RasterLite2 Styles

avoiding the need to redefine the functions parameters of all the intermediate function being called.

Now that Sandro has resumed the final stage of the Rasterlite2 development
* this will become important if Rasterlite2 Styles are to be supported

**Comment by moovida** [comment]
closing as conflicting
